airbyte-cdk/java/airbyte-cdk/dependencies/src/main/kotlin/io/airbyte/commons/json/Jsons.kt

```diff
@@ -262,16 +262,16 @@ object Jsons {
     }
 
     fun navigateTo(node: JsonNode, keys: List<String?>): JsonNode {
-        var node = node
+        var targetNode = node
         for (key in keys) {
-            node = node[key]
+            targetNode = targetNode[key]
         }
-        return node
+        return targetNode
     }
 
     fun replaceNestedValue(json: JsonNode, keys: List<String?>, replacement: JsonNode?) {
         replaceNested(json, keys) { node: ObjectNode, finalKey: String? ->
-            node.put(finalKey, replacement)
+            node.replace(finalKey, replacement)
```

Comment:
I hope this doesn't break existing behaviour! `replace` is more strict than `put`. Still, I approve of this change anyway.